zebra-chain/src/transaction/arbitrary.rs

```diff
@@ -218,59 +220,168 @@ impl Transaction {
         }
     }
 
-    /// Fixup non-coinbase transparent values and shielded value balances,
-    /// so that this transaction passes the "remaining transaction value pool" check.
+    /// Fixup transparent values and shielded value balances,
+    /// so that transaction and chain value pools won't overflow MAX_MONEY.
     ///
-    /// Returns the remaining transaction value.
+    /// These fixes are applied to coinbase and non-coinbase transactions.
```

Comment:
This diff might be easier to review in Split diff mode.
(This setting is under the ⚙️ icon at the top of the file view.)

https://github.blog/2014-09-03-introducing-split-diffs/